pkg/operator/ceph/cluster/osd/health.go

```diff
@@ -90,29 +80,46 @@ func (m *Monitor) osdStatus() error {
 		id := int(id64)
 
 		logger.Debugf("validating status of osd.%d", id)
-		_, tracked := m.lastStatus[id]
 
-		// No action on in/out cluster state is taken at this time.
-		status, _, err := osdDump.StatusByID(int64(id))
+		status, in, err := osdDump.StatusByID(int64(id))
 		if err != nil {
 			return err
 		}
 
 		if status != upStatus {
 			logger.Infof("osd.%d is marked 'DOWN'", id)
-			if tracked {
-				evalDownStatus(id)
-			} else {
-				m.lastStatus[id] = time.Now()
-			}
 		} else {
 			logger.Debugf("osd.%d is healthy.", id)
-			if tracked {
-				logger.Infof("osd.%d recovered, stopping tracking.", id)
-				delete(m.lastStatus, id)
+
+		}
+
+		if in != inStatus {
+			logger.Debugf("osd.%d is marked 'OUT'", id)
+			if err := m.handleOSDMarkedOut(id); err != nil {
+				return err
```

Comment:
How about if we just log this error instead of return it? This is best effort to clean up the OSDs. A single failure to cleanup shouldn't block other OSDs from cleaning up.

Reply:
Updated